### PR TITLE
docs: update and fix examples for `no-unused-vars`

### PR DESCRIPTION
**Pull request by Tanujkanti4441**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
some examples inside the `correct` tag of `no-unused-vars` rule docs are showing errors, so just fixed them by adding some relevant code and updated the docs page a bit.

#### Is there anything you'd like reviewers to focus on?
[Example1 in playground](https://eslint.org/play/#eyJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsic291cmNlVHlwZSI6Im1vZHVsZSJ9fSwidGV4dCI6Ii8qZXNsaW50IG5vLXVudXNlZC12YXJzOiBcImVycm9yXCIqL1xuXG52YXIgeCA9IDEwO1xuYWxlcnQoeCk7XG5cbi8vIGZvbyBpcyBjb25zaWRlcmVkIHVzZWQgaGVyZVxubXlGdW5jKGZ1bmN0aW9uIGZvbygpIHtcbiAgICAvLyAuLi5cbn0uYmluZCh0aGlzKSk7XG5cbihmdW5jdGlvbihmb28pIHtcbiAgICByZXR1cm4gZm9vO1xufSkoKTtcblxudmFyIG15RnVuYztcbm15RnVuYyA9IHNldFRpbWVvdXQoZnVuY3Rpb24oKSB7XG4gICAgLy8gbXlGdW5jIGlzIGNvbnNpZGVyZWQgdXNlZFxuICAgIG15RnVuYygpO1xufSwgNTApO1xuXG4vLyBPbmx5IHRoZSBzZWNvbmQgYXJndW1lbnQgZnJvbSB0aGUgZGVzdHJ1Y3R1cmVkIGFycmF5IGlzIHVzZWQuXG5mdW5jdGlvbiBnZXRZKFssIHldKSB7XG4gICAgcmV0dXJuIHk7XG59In0=)
[Example2 in playground](https://eslint.org/play/#eyJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsic291cmNlVHlwZSI6Im1vZHVsZSJ9fSwidGV4dCI6Ii8qZXNsaW50IG5vLXVudXNlZC12YXJzOiBbXCJlcnJvclwiLCB7IFwiaWdub3JlUmVzdFNpYmxpbmdzXCI6IHRydWUgfV0qL1xuLy8gJ2ZvbycgYW5kICdiYXInIHdlcmUgaWdub3JlZCBiZWNhdXNlIHRoZXkgaGF2ZSBhIHJlc3QgcHJvcGVydHkgc2libGluZy5cbnZhciB7IGZvbywgLi4uY29vcmRzIH0gPSBkYXRhO1xuXG52YXIgYmFyO1xuKHsgYmFyLCAuLi5jb29yZHMgfSA9IGRhdGEpOyJ9)

### CODE REVIEW

File: docs/src/rules/no-unused-vars.md [hunk 1]
```diff
@@ -105,23 +107,45 @@ Note that `/* exported */` has no effect for any of the following:
...
+    "rules": {
+        "no-unused-vars": ["error", "local"]
+    }
+}
+```
+
+Object setting has several options.
 
 ```json
 {
```
Comment (mdjermanovic) [inline docs/src/rules/no-unused-vars.md:151]: I think this section was fine and seems better to leave it as is. The description of string settings refers to the `vars` property which is explained in detail below (note that `"local"` allows unused global variables, and neither of these two allows unused local variables, so the added description might be confusing). The config below is intended to show defaults for this rule, not an example of the object option.

File: docs/src/rules/no-unused-vars.md [hunk 2]
```diff
@@ -105,23 +107,45 @@ Note that `/* exported */` has no effect for any of the following:
...
 The line comment `// exported variableName` will not work as `exported` is not line-specific.
 
-Examples of **correct** code for `/* exported variableName */` operation:
+```js
+/* exported global_var */
+
+var global_var = 42;
+```
+
+Examples of **correct** code for `/* exported variableName */` operation with `no-unused-var`:
```
Comment (mdjermanovic) [inline docs/src/rules/no-unused-vars.md:116]:
```suggestion
Examples of **correct** code for `/* exported variableName */` operation with `no-unused-vars`:
```

File: docs/src/rules/no-unused-vars.md [hunk 3]
```diff
@@ -105,23 +107,45 @@ Note that `/* exported */` has no effect for any of the following:
...
 
 ::: correct
```
Comment (mdjermanovic) [inline docs/src/rules/no-unused-vars.md:118]:
```suggestion
::: correct { "sourceType": "script" }
```

We can add this config to make the example valid in the Playground.

File: docs/src/rules/no-unused-vars.md [hunk 4]
```diff
@@ -105,23 +107,45 @@ Note that `/* exported */` has no effect for any of the following:
...
 
 ```js
+/*eslint no-unused-vars: "error"*/
 /* exported global_var */
 
+// rule will not report global_var
+// sourcType must be set to script
+// environment should not be node or commonjs
+// globalReturn must be set to false
+
```
Comment (mdjermanovic) [inline docs/src/rules/no-unused-vars.md:128]:
```suggestion
```

I think there's no need for this as the environment (and thus implicitly the ESLint configuration) where this applies is already explained at the beginning of this section.

**Review by mdjermanovic — CHANGES_REQUESTED** [review]

File: docs/src/rules/no-unused-vars.md (hunk 4 above)
Comment (Tanujkanti4441) [reply docs/src/rules/no-unused-vars.md]: Done!

**Comment by Tanujkanti4441** [comment]
@mdjermanovic, have got few more rule docs that have such examples, is it ok if i fix them in this PR, or it would be good to open a new one?

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Comment by Tanujkanti4441** [comment]
> > have found few more rule docs that have such examples, is it ok if i fix them in this PR, or it would be good to open a new one?
> 
> Since this is finished, let's merge it and fix other rules in separate PR(s).

Ok thanks!

**Pull request merged**